fw/sock.c

```diff
@@ -805,7 +820,12 @@ ss_tcp_process_skb(struct sock *sk, struct sk_buff *skb, int *processed)
 		 */
 		BUG_ON(conn == NULL);
 
-		if (SS_CONN_TYPE(sk) & Conn_Stop) {
+		/*
+		 * We should continue to proess WINDOW_UPDATE frames
+		 * for closing connections.
+		 */
+		if ((SS_CONN_TYPE(sk) & Conn_Stop)
+		    && !((SS_CONN_TYPE(sk) & Conn_H2Clnt) == Conn_H2Clnt)) {
```

Comment:
It's quite bad that the low level socket API must know about HTTP/2, which is 2 layers higher. Is this absolutely necessary?

Reply:
Fixed